Sources/Purchasing/ProductsManager.swift

```diff
@@ -71,6 +71,17 @@ class ProductsManager: NSObject {
         productsFetcherSK1.cacheProduct(product)
     }
 
+    func clearCachedProducts() {
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *),
+           self.systemInfo.storeKit2Setting == .enabledForCompatibleDevices {
+            Task<Void, Never> {
+                await productsFetcherSK2.clearCache()
```

Comment:
I just realized that `ProductsFetcherSK2` was already keeping track of the `Storefront` and clearing the cache itself, so this now feels like duplicated logic. Perhaps the best solution is to remove that custom logic from within `ProductsFetcherSK2`?

Reply:
Yep, you're right, the worst thing is we keep the outdated products in the cache, but I think it's not a real problem since we're going to re-fetch the product when the Storefront changes in the next step 👍🏼 

Let me change this code 👍🏼

Reply:
Done ✅ ([38ca1e7](https://github.com/RevenueCat/purchases-ios/pull/1583/commits/38ca1e79af91ee07033aac18ece05158018afe28))

Reply:
Oh looks like you opted for leaving only the internal cache-clearing implementation?
What do you think of doing it the other way around for consistency with SK1?

Reply:
mm...sorry, I don't know exactly what you mean 🤔

Reply:
I mentioned this earlier:
> `ProductsFetcherSK2` was already keeping track of the Storefront and clearing the cache itself, so this now feels like duplicated logic. 

So you removed the logic from `ProductsManager`. I was thinking that maybe instead it made more sense to make `ProductsManager` clear the cache like you had before, and instead remove the current `Storefront` keeping logic from `ProductsFetcherSK2`, so it's consistent with how `SK1`'s fetcher works.

Reply:
👋🏼  I was trying to do the change but I found out that you put [a note in the code to warn about using `Storefront.updates`](https://github.com/RevenueCat/purchases-ios/blob/395322ac7ef437f6b2f880b2057559b11954bdc4/Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift#L94) could cause race conditions ([original commit](https://github.com/RevenueCat/purchases-ios/commit/e9a8dcf8076a7b2463dfdd789c8b635f5ddbc934)) 🤔 

Do you think we could have some problems using `Storefront` updates in production or it's something related to the tests?

Reply:
Ey guys, Did you guys finally get a look at this? 🤔 Thanks!

Reply:
@NachoSoto I think this got lost in your notifications 😄